zha/application/platforms/sensor/__init__.py

```diff
@@ -201,6 +201,21 @@ def __init__(
             self.handle_cluster_handler_attribute_updated,
         )
 
+    def _validate_state_class(
+        self,
+        state_class_value: SensorStateClass,
+    ) -> SensorStateClass | None:
+        """Validate and return a state class."""
+        try:
+            return SensorStateClass(state_class_value.value)
+        except ValueError as ex:
+            _LOGGER.warning(
+                "Quirks provided an invalid state class: %s: %s",
+                state_class_value,
+                ex,
+            )
+            return None
+
```

Comment:
Even though this is only used for the sensor platform, I feel like we could still move this into the [helpers module](https://github.com/zigpy/zha/blob/dev/zha/application/platforms/helpers.py) (as a function), due to it being very similar to the device class validation and to avoid cluttering the sensor class more. wdyt?

Reply:
I've placed it in the `helpers` module initially, but moved it to `sensor` because this imports `SensorStateClass` (which is in the `sensor` module) and resulted in a circular import.